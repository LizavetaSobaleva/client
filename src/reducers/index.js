import { applyMiddleware, combineReducers } from "redux";
import { configureStore} from "@reduxjs/toolkit"
import { thunk } from "redux-thunk";
import userReducer from "./userReducer";
import fileReducer from "./fileReducer";
import uploadReducer from "./uploadReducer";
import appReducer from "./appReducer";
import { composeWithDevTools } from '@redux-devtools/extension';


const rootReducer = combineReducers({
    user: userReducer,
    files: fileReducer,
    upload: uploadReducer,
    app: appReducer
})

export const store = configureStore({reducer: rootReducer, devTools: composeWithDevTools(applyMiddleware(thunk))})