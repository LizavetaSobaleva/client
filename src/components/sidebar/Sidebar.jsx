import React, { useState } from 'react'
import './sidebar.less'
import { useDispatch } from 'react-redux'
import { logout } from '../../reducers/userReducer'
import Logo from '../../assets/img/logo.png'
import Logout from '../../assets/img/logout.svg'
import SecondaryButton from '../UI/secondaryButton/SecondaryButton'
import { NavLink } from 'react-router-dom'
import Navigation from './navigation/Navigation'
import DropdownButton from '../UI/dropdownButton/DropdownButton'


const Sidebar = () => {
    const dispatch = useDispatch()
    const [isOpen, setIsOpen] = useState(false)

    const toggleDropdown = () => {
      setIsOpen(!isOpen)
    }

    window.addEventListener('resize', function() {
      const screenWidth = window.innerWidth
      const sidebar = document.querySelector('.sidebar')
    
      if (screenWidth <= 768) {
        sidebar.classList.add('compressed')
      } else {
        sidebar.classList.remove('compressed')
      }
    })
    


  return (
    <div className={`sidebar ${isOpen ? 'compressed' : ''}`}>
        <div className="sidebar__content">
          <NavLink to="/">
            <div className="sidebar__header">
              <img src={Logo} alt="" className="sidebar__logo"/>
              <div className="sidebar__title">MERN cloud</div>
            </div>
          </NavLink>
          <div className="sidebar__navigation">
            <Navigation/>
          </div>
          <div className="sidebar__logout" onClick={() => dispatch(logout())}>
            
            <SecondaryButton>
              <img src={Logout} alt="" className="sidebar__logout-icon"/>
              <div className="sidebar__logout-text">Log out</div>
            </SecondaryButton>
          </div>
        </div>
        <div className='sidebar__compress' onClick={toggleDropdown}>
          <DropdownButton/>
        </div>
    </div>
  )
}

export default Sidebar