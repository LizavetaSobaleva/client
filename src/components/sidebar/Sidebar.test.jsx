import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import Sidebar from './Sidebar';
import { logout } from '../../reducers/userReducer';

const mockStore = configureStore([]);
let store;

const renderComponent = (store) =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Sidebar />
      </MemoryRouter>
    </Provider>
  );

describe('Sidebar Component', () => {
  beforeEach(() => {
    store = mockStore({
      user: { isAuth: true, currentUser: { name: 'John Doe', avatar: null } },
      files: { dirStructure: [{ _id: '1', name: 'Dir 1', path: 'dir1', idPath: ['1'], childs: [] }] }
    });
    store.dispatch = jest.fn();
  });

  test('renders Sidebar without crashing', () => {
    renderComponent(store);
    expect(screen.getByTestId('sidebar')).toBeInTheDocument();
    expect(screen.getByTestId('sidebarContent')).toBeInTheDocument();
  });

  test('renders sidebar header with logo and title', () => {
    renderComponent(store);
    expect(screen.getByTestId('sidebarHeader')).toBeInTheDocument();
    expect(screen.getByTestId('sidebarLogo')).toBeInTheDocument();
    expect(screen.getByText('MERN cloud')).toBeInTheDocument();
  });

  test('renders navigation and logout button', () => {
    renderComponent(store);
    expect(screen.getByTestId('sidebarNavigation')).toBeInTheDocument();
    expect(screen.getByTestId('sidebarLogout')).toBeInTheDocument();
  });

  test('logout button triggers logout action', () => {
    renderComponent(store);
    fireEvent.click(screen.getByTestId('logoutButton'));
    expect(store.dispatch).toHaveBeenCalledWith(logout());
  });

  test('toggles sidebar compression on button click', () => {
    renderComponent(store);
    const sidebar = screen.getByTestId('sidebar');
    const compressButton = screen.getByTestId('compressButton');
    expect(sidebar).not.toHaveClass('compressed');
    fireEvent.click(compressButton);
    expect(sidebar).toHaveClass('compressed');
    fireEvent.click(compressButton);
    expect(sidebar).not.toHaveClass('compressed');
  });
});
