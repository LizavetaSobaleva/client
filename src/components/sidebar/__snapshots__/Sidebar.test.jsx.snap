// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component renders Sidebar without crashing 1`] = `
<DocumentFragment>
  <div
    class="sidebar "
    data-testid="sidebar"
  >
    <div
      class="sidebar__content"
      data-testid="sidebarContent"
    >
      <a
        aria-current="page"
        class="active"
        data-testid="homeLink"
        href="/"
      >
        <div
          class="sidebar__header"
          data-testid="sidebarHeader"
        >
          <img
            alt=""
            class="sidebar__logo"
            data-testid="sidebarLogo"
            src="test-file-stub"
          />
          <div
            class="sidebar__title"
          >
            MERN cloud
          </div>
        </div>
      </a>
      <div
        class="sidebar__navigation"
        data-testid="sidebarNavigation"
      >
        <div
          class="navigation"
          data-testid="navigation"
        >
          <div
            class="navigation__item"
            data-testid="navigationItem"
          >
            <a
              aria-current="page"
              class="active"
              data-testid="navLink"
              href="/"
            >
              <div
                class="navigation__title"
                data-testid="navigationTitle"
              >
                <img
                  alt=""
                  class="navigation__icon"
                  data-testid="navigationIcon"
                  src="test-file-stub"
                />
                <div
                  class="navigation__name"
                >
                  My Cloud
                </div>
                <div
                  class="navigation__dropdown"
                  data-testid="navigationDropdown"
                >
                  <div
                    class="dropdown"
                    data-testid="dropdownButton"
                  >
                    <img
                      alt=""
                      class="dropdown__icon "
                      data-testid="dropdownIcon"
                      src="test-file-stub"
                    />
                  </div>
                </div>
              </div>
            </a>
            <div
              class="navigation__content visible"
              data-testid="navigationContent"
            >
              <div
                class="navigation__element"
                data-testid="navigationElement"
              >
                <div
                  class="diritem"
                  data-testid="dirItem-1"
                >
                  <div
                    class="diritem__content"
                  >
                    <div
                      class="diritem__title"
                      data-testid="dirItemTitle-1"
                    >
                      Dir 1
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sidebar__logout"
        data-testid="sidebarLogout"
      >
        <button
          class="secondaryButton"
          data-testid="logoutButton"
        >
          <img
            alt=""
            class="sidebar__logout-icon"
            data-testid="logoutIcon"
            src="test-file-stub"
          />
          <div
            class="sidebar__logout-text"
          >
            Log out
          </div>
        </button>
      </div>
    </div>
    <div
      class="sidebar__compressBtn"
      data-testid="compressButton"
    >
      <div
        class="dropdown"
        data-testid="dropdownButton"
      >
        <img
          alt=""
          class="dropdown__icon "
          data-testid="dropdownIcon"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sidebar Component renders navigation and logout button 1`] = `
<DocumentFragment>
  <div
    class="sidebar "
    data-testid="sidebar"
  >
    <div
      class="sidebar__content"
      data-testid="sidebarContent"
    >
      <a
        aria-current="page"
        class="active"
        data-testid="homeLink"
        href="/"
      >
        <div
          class="sidebar__header"
          data-testid="sidebarHeader"
        >
          <img
            alt=""
            class="sidebar__logo"
            data-testid="sidebarLogo"
            src="test-file-stub"
          />
          <div
            class="sidebar__title"
          >
            MERN cloud
          </div>
        </div>
      </a>
      <div
        class="sidebar__navigation"
        data-testid="sidebarNavigation"
      >
        <div
          class="navigation"
          data-testid="navigation"
        >
          <div
            class="navigation__item"
            data-testid="navigationItem"
          >
            <a
              aria-current="page"
              class="active"
              data-testid="navLink"
              href="/"
            >
              <div
                class="navigation__title"
                data-testid="navigationTitle"
              >
                <img
                  alt=""
                  class="navigation__icon"
                  data-testid="navigationIcon"
                  src="test-file-stub"
                />
                <div
                  class="navigation__name"
                >
                  My Cloud
                </div>
                <div
                  class="navigation__dropdown"
                  data-testid="navigationDropdown"
                >
                  <div
                    class="dropdown"
                    data-testid="dropdownButton"
                  >
                    <img
                      alt=""
                      class="dropdown__icon "
                      data-testid="dropdownIcon"
                      src="test-file-stub"
                    />
                  </div>
                </div>
              </div>
            </a>
            <div
              class="navigation__content visible"
              data-testid="navigationContent"
            >
              <div
                class="navigation__element"
                data-testid="navigationElement"
              >
                <div
                  class="diritem"
                  data-testid="dirItem-1"
                >
                  <div
                    class="diritem__content"
                  >
                    <div
                      class="diritem__title"
                      data-testid="dirItemTitle-1"
                    >
                      Dir 1
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sidebar__logout"
        data-testid="sidebarLogout"
      >
        <button
          class="secondaryButton"
          data-testid="logoutButton"
        >
          <img
            alt=""
            class="sidebar__logout-icon"
            data-testid="logoutIcon"
            src="test-file-stub"
          />
          <div
            class="sidebar__logout-text"
          >
            Log out
          </div>
        </button>
      </div>
    </div>
    <div
      class="sidebar__compressBtn"
      data-testid="compressButton"
    >
      <div
        class="dropdown"
        data-testid="dropdownButton"
      >
        <img
          alt=""
          class="dropdown__icon "
          data-testid="dropdownIcon"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sidebar Component renders sidebar header with logo and title 1`] = `
<DocumentFragment>
  <div
    class="sidebar "
    data-testid="sidebar"
  >
    <div
      class="sidebar__content"
      data-testid="sidebarContent"
    >
      <a
        aria-current="page"
        class="active"
        data-testid="homeLink"
        href="/"
      >
        <div
          class="sidebar__header"
          data-testid="sidebarHeader"
        >
          <img
            alt=""
            class="sidebar__logo"
            data-testid="sidebarLogo"
            src="test-file-stub"
          />
          <div
            class="sidebar__title"
          >
            MERN cloud
          </div>
        </div>
      </a>
      <div
        class="sidebar__navigation"
        data-testid="sidebarNavigation"
      >
        <div
          class="navigation"
          data-testid="navigation"
        >
          <div
            class="navigation__item"
            data-testid="navigationItem"
          >
            <a
              aria-current="page"
              class="active"
              data-testid="navLink"
              href="/"
            >
              <div
                class="navigation__title"
                data-testid="navigationTitle"
              >
                <img
                  alt=""
                  class="navigation__icon"
                  data-testid="navigationIcon"
                  src="test-file-stub"
                />
                <div
                  class="navigation__name"
                >
                  My Cloud
                </div>
                <div
                  class="navigation__dropdown"
                  data-testid="navigationDropdown"
                >
                  <div
                    class="dropdown"
                    data-testid="dropdownButton"
                  >
                    <img
                      alt=""
                      class="dropdown__icon "
                      data-testid="dropdownIcon"
                      src="test-file-stub"
                    />
                  </div>
                </div>
              </div>
            </a>
            <div
              class="navigation__content visible"
              data-testid="navigationContent"
            >
              <div
                class="navigation__element"
                data-testid="navigationElement"
              >
                <div
                  class="diritem"
                  data-testid="dirItem-1"
                >
                  <div
                    class="diritem__content"
                  >
                    <div
                      class="diritem__title"
                      data-testid="dirItemTitle-1"
                    >
                      Dir 1
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sidebar__logout"
        data-testid="sidebarLogout"
      >
        <button
          class="secondaryButton"
          data-testid="logoutButton"
        >
          <img
            alt=""
            class="sidebar__logout-icon"
            data-testid="logoutIcon"
            src="test-file-stub"
          />
          <div
            class="sidebar__logout-text"
          >
            Log out
          </div>
        </button>
      </div>
    </div>
    <div
      class="sidebar__compressBtn"
      data-testid="compressButton"
    >
      <div
        class="dropdown"
        data-testid="dropdownButton"
      >
        <img
          alt=""
          class="dropdown__icon "
          data-testid="dropdownIcon"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
